### PR TITLE
Improvement: Tp to nearest infested plot command

### PR DESCRIPTION
**Pull request by ItsEmpa**
## What
Adds a `/shtpinfested` command to tp to nearest infested plot.

[Discord Suggestion](https://ptb.discord.com/channels/997079228510117908/1238591490784104659)

## Changelog Improvements
+ Added `/shtpinfested` command to teleport to nearest infested plot. - Empa

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt [hunk 1]
```diff
@@ -165,6 +165,18 @@ class PestFinder {
...
 
+        teleportNearestInfectedPlot()
+    }
+
+    @SubscribeEvent
+    fun onItemInHandChange(event: ItemInHandChangeEvent) {
+        if (!isEnabled()) return
+        if (!config.showPlotInWorld) return
+        if (event.oldItem !in PestAPI.vacuumVariants) return
+        lastTimeVacuumHold = SimpleTimeMark.now()
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt]: this should go into pest api

**Review by hannibal002 — DISMISSED** [review]

File: src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt (hunk 1 above)
Comment (ItsEmpa) [reply src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt]: you mean the entire onKeyClick event? or just the teleportNearestInfectedPlot function

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt]: the value lastTimeVacuumHold and the logic that calculates it

Comment (ItsEmpa) [reply src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt]: ah, looked at the wrong thing.

**Review by CalMWolfs — APPROVED** [review]

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**